tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -405,9 +405,54 @@ bool IsSupportedActivation(const NodeDef& node) {
   return (is_default_supported || is_mkl_specific);
 }
 
+bool RuntimeFusionEnabled(const Cluster* cluster) {
+  static bool is_enabled = [&] {
+#if CUDNN_VERSION >= 8400
+    // Cudnn runtime fusion feature is recommended for Ampere GPUs or later.
+    // For pre-Ampere GPUs, the overhead of runtime compilation would be very
+    // large and there are more limitations of supported cases.
+    if (!cluster) return false;
+    auto devices = cluster->GetDevices();
+    int num_gpus = 0;
+    int num_ampere = 0;
+    for (const auto& d : devices) {
+      if (d.second.type() == "GPU") {
+        num_gpus++;
+        auto cc_it = d.second.environment().find("architecture");
+        if (cc_it != d.second.environment().end()) {
+          double compute_capability = 0.0;
+          if (absl::SimpleAtod(cc_it->second, &compute_capability) &&
+              compute_capability >= 8.0) {
+            num_ampere++;
+          }
+        }
+      }
+    }
+    bool runtime_fusion_enabled = CudnnUseRuntimeFusion() &&
+                                  CudnnUseFrontend() && num_gpus > 0 &&
+                                  num_gpus == num_ampere;
+
+    if (CudnnUseRuntimeFusion() && !runtime_fusion_enabled) {
+      VLOG(WARNING) << "Enabling Cudnn with runtime compilation requires the "
+                    << "Cudnn frontend and Ampere GPUs or later, but we got "
+                    << "Cudnn frontend is "
+                    << (CudnnUseFrontend() ? "enabled" : "disabled") << " and "
+                    << num_ampere << " Ampere GPU(s) out of total " << num_gpus
+                    << " GPU(s)";
+    }
```

Comment:
Much of this logic seems to be copy-pasted from the equivalent function in StreamExecutor.

Would it be worthwhile to common the logic?

Reply:
I already started creating a CL for this PR and copying the internal changes from the previous rolled-back CL into the new one. If the logic should be made common, can this be done in a new PR?

Reply:
Commoning the logic in a separate CL sounds fine to me.